internal/backend/runtime/omni/migration/migration_test.go

```diff
@@ -1798,6 +1799,183 @@ func (suite *MigrationSuite) TestMoveEtcdBackupStatuses() {
+  features:
+    foo: bar
+    apidCheckExtKeyUsage: true
+    baz: qux
+    diskQuotaSupport: true
+    quux: corge
+`)))
+
+	cluster12configStatus3 := omni.NewClusterMachineConfigStatus(resources.DefaultNamespace, "cluster-12-machine-3")
+	cluster12configStatus3.TypedSpec().Value.ClusterMachineConfigVersion = "1" // mark the v1 to be applied
+
+	// patch expected: **only** diskQuotaSupport needs to be preserved
+	cluster14config1 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-1")
+	cluster14config1.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config1.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+    foo: bar
+    diskQuotaSupport: true
+    apidCheckExtKeyUsage: true
+`)))
+
+	cluster14configStatus1 := omni.NewClusterMachineConfigStatus(resources.DefaultNamespace, "cluster-14-machine-1")
+	cluster14configStatus1.TypedSpec().Value.ClusterMachineConfigVersion = "1" // mark the v1 to be applied
+
+	// no patch expected: it does not have any of the features to be preserved
+	cluster14config2 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-2")
+	cluster14config2.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config2.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+    foo: bar
+`)))
+
+	cluster14configStatus2 := omni.NewClusterMachineConfigStatus(resources.DefaultNamespace, "cluster-14-machine-2")
+	cluster14configStatus2.TypedSpec().Value.ClusterMachineConfigVersion = "1" // mark the v1 to be applied
+
+	// no patch expected, since it will be in tearing down phase
+	cluster14config3 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-3")
+	cluster14config3.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config3.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+    foo: bar
+    diskQuotaSupport: true
+`)))
+
+	cluster14configStatus3 := omni.NewClusterMachineConfigStatus(resources.DefaultNamespace, "cluster-14-machine-3")
+	cluster14configStatus3.TypedSpec().Value.ClusterMachineConfigVersion = "1" // mark the v1 to be applied
+
+	// no patch expected, since the ClusterMachineConfigStatus.Spec.ClusterMachineConfigVersion will indicate that this version of config was not applied
+	cluster14config4 := omni.NewClusterMachineConfig(resources.DefaultNamespace, "cluster-14-machine-4")
+	cluster14config4.Metadata().Labels().Set(omni.LabelCluster, "cluster-14")
+
+	suite.Require().NoError(cluster14config4.TypedSpec().Value.SetUncompressedData([]byte(`machine:
+  features:
+    foo: bar
+    diskQuotaSupport: true
+`)))
+
+	cluster14configStatus4 := omni.NewClusterMachineConfigStatus(resources.DefaultNamespace, "cluster-14-machine-4")
+	cluster14configStatus4.TypedSpec().Value.ClusterMachineConfigVersion = "0" // mark the v0 to be applied, but v1 was not applied
```

Comment:
This simulates a config which was generated but not applied (e.g., a disconnected machine).
We assert that it doesn't get the patch.